### PR TITLE
Adding VerifyClientMessage to ClientState interface

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #1191 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/core/02-client/legacy/v100/solomachine.go [hunk 1]
```diff
@@ -88,6 +88,13 @@ func (cs ClientState) ExportMetadata(_ sdk.KVStore) []exported.GenesisMetadata {
 	panic("legacy solo machine is deprecated!")
 }
 
+// VerifyClientMessage returns nil
+func (cs *ClientState) VerifyClientMessage(
```
Comment (seantking) [inline modules/core/02-client/legacy/v100/solomachine.go:92]: I wasn't too sure what we wanted to do here (if anything). 

What is the history behind this legacy solo machine anyway? :)

**Review by damiannolan — COMMENTED** [review]
There was type casting in tests because `VerifyClientMessage` wasn't an interface function.
I think you can remove the TODOs are part of this PR. 

I think its just these two tests:
https://github.com/cosmos/ibc-go/blob/02-client-refactor/modules/light-clients/07-tendermint/types/update_test.go#L385
https://github.com/cosmos/ibc-go/blob/02-client-refactor/modules/light-clients/07-tendermint/types/misbehaviour_handle_test.go#L779

**Comment by seantking** [comment]
> There was type casting in tests because `VerifyClientMessage` wasn't an interface function. I think you can remove the TODOs are part of this PR.
> 
> I think its just these two tests: https://github.com/cosmos/ibc-go/blob/02-client-refactor/modules/light-clients/07-tendermint/types/update_test.go#L385 https://github.com/cosmos/ibc-go/blob/02-client-refactor/modules/light-clients/07-tendermint/types/misbehaviour_handle_test.go#L779

Good catch :+1: Done

File: modules/core/02-client/legacy/v100/solomachine.go [hunk 2]
```diff
@@ -88,6 +88,13 @@ func (cs ClientState) ExportMetadata(_ sdk.KVStore) []exported.GenesisMetadata {
 	panic("legacy solo machine is deprecated!")
 }
 
+// VerifyClientMessage panics
```
Comment (damiannolan) [inline modules/core/02-client/legacy/v100/solomachine.go:91]: Mega nit, add the `!` to godoc for consistency

```suggestion
// VerifyClientMessage panics!
```

**Review by damiannolan — APPROVED** [review]

File: modules/core/exported/client.go [hunk 3]
```diff
@@ -56,8 +56,10 @@ type ClientState interface {
 	// Genesis function
 	ExportMetadata(sdk.KVStore) []GenesisMetadata
 
-	// Update and Misbehaviour functions
+	// VerifyClientMessage verifies a Header or a Misbehaviour
```
Comment (colin-axner) [inline modules/core/exported/client.go:59]:
```suggestion
	// VerifyClientMessage verifies a ClientMessage. A ClientMessage could be a Header, Misbehaviour, or batch update.
```
we should probably add more description here

**Review by colin-axner — APPROVED** [review]
Wonderful!

File: modules/core/exported/client.go (hunk 3 above)
Comment (seantking) [reply modules/core/exported/client.go:59]: Yeah much better thanks

**Pull request merged**